core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -186,44 +73,16 @@ private RpcUtils() {} // prevent instantiation
   public static <T> void call(Logger logger, RpcCallableThrowsIOException<T> callable,
       String methodName, boolean failureOk, String description, StreamObserver<T> responseObserver,
       Object... args) {
-    // avoid string format for better performance if debug is off
-    String debugDesc = logger.isDebugEnabled() ? String.format(description, args) : null;
-    try (Timer.Context ctx = MetricsSystem.timer(getQualifiedMetricName(methodName)).time()) {
-      logger.debug("Enter: {}: {}", methodName, debugDesc);
-      T ret = callable.call();
-      logger.debug("Exit (OK): {}: {}", methodName, debugDesc);
-      responseObserver.onNext(ret);
-      responseObserver.onCompleted();
-    } catch (AlluxioException e) {
-      logger.debug("Exit (Error): {}: {}", methodName, debugDesc, e);
-      if (!failureOk) {
-        MetricsSystem.counter(getQualifiedFailureMetricName(methodName)).inc();
-        if (!logger.isDebugEnabled()) {
-          logger.warn("Exit (Error): {}: {}, Error={}", methodName,
-              String.format(description, args), e);
-        }
-      }
+    try {
       responseObserver
-          .onError(AlluxioStatusException.fromAlluxioException(e).toGrpcStatusException());
-    } catch (IOException e) {
-      logger.debug("Exit (Error): {}: {}", methodName, debugDesc, e);
-      if (!failureOk) {
-        MetricsSystem.counter(getQualifiedFailureMetricName(methodName)).inc();
-        if (!logger.isDebugEnabled()) {
-          logger.warn("Exit (Error): {}: {}, Error={}", methodName,
-              String.format(description, args), e);
-        }
-      }
-      responseObserver.onError(AlluxioStatusException.fromIOException(e).toGrpcStatusException());
-    } catch (RuntimeException e) {
-      logger.error("Exit (Error): {}: {}", methodName, String.format(description, args), e);
-      MetricsSystem.counter(getQualifiedFailureMetricName(methodName)).inc();
-      responseObserver.onError(new InternalException(e).toGrpcStatusException());
+          .onNext(callAndReturn(logger, callable, methodName, failureOk, description, args));
+      responseObserver.onCompleted();
```

Comment:
Move this line out of the try-catch to avoid error due to calling onError on closed call?

Reply:
Did some more hardening.